git_repo/services/ext/gitbucket.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+
+import logging
+log = logging.getLogger('git_repo.gitbucket')
+
+from ..service import register_target
+from .github import GithubService
+
+import github3
+
+@register_target('bucket', 'gitbucket')
+class GitbucketService(GithubService):
+    fqdn = "localhost"
+    port = 8080
+    def __init__(self, *args, **kwarg):
+        super(GitbucketService, self).__init__(*args, **kwarg)
+
+    def connect(self):
+        self.gh = github3.GitHubEnterprise(self.build_url(self))
```

Comment:
ok, I'll make a ticket to github3.py for this issue, so he splits the init in several steps. (by having a default URL to an empty invalid value, and a `set_url()` method)


Currently, here's the [code it is doing](https://github.com/sigmavirus24/github3.py/blob/0.9.6/github3/github.py#L1490,L1493) at startup time.

So we keep my suggestion above, but the `connect()` method shall change:

```python
def connect(self):
    self.gh._session.base_url = url.rstrip('/') + '/api/v3'
    self.gh._session.verify = self.session_certificate or not self.session_insecure
    super(GitbucketService, self).connect()
```